### PR TITLE
Using udev rule to change Netdev name

### PR DESCRIPTION
**Pull request by Mmduh-483**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -231,6 +231,23 @@ The device plugin will initially discover all PCI network resources in the host
...
+# vi vi /etc/udev/rules.d/new.rules
+SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", KERNELS=="0000:02:00.0", NAME="pf0s0f1"
+``` 
+`KERNELS` is the pci address for the NIC
+`NAME` is the new name for the NIC
+
+2- Run udev commands
+```
+# udevadm control --reload
+# udevadm control --reload
```
Comment (moshe010) [inline README.md]: Why you have it twice?

Comment (Mmduh-483) [reply README.md]: fixed

File: README.md [hunk 2]
```diff
@@ -231,6 +231,24 @@ The device plugin will initially discover all PCI network resources in the host
 3. "drivers" - The driver name the device is registered with
 4. "pfNames" - The Physical funtion name
 
+To change the NIC name using the udev rules:
```
Comment (zshi-redhat) [inline README.md]: Might be worth mentioning why we are adding udev rule settings here, for example, it's for better use of the `pfNames` filter across cluster nodes, to configure identical interface names for devices on different nodes.

File: README.md [hunk 3]
```diff
@@ -239,6 +239,26 @@ The device plugin will initially discover all PCI network resources in the host
 4. "pfNames" - The Physical funtion name
 5. "linkTypes" - The link type of the net device associated with the PCI device.
 
+### Using UDEV Rules forchanging NIC's name:
```
Comment (zshi-redhat) [inline README.md]: add space between `for and changing`

File: README.md [hunk 4]
```diff
@@ -239,6 +239,26 @@ The device plugin will initially discover all PCI network resources in the host
...
+
+UDEV rules allow the user to change the NIC's name every time the node get rebooted:
+
+1. Add a new rule:
+```
+# vi vi /etc/udev/rules.d/new.rules
```
Comment (zshi-redhat) [inline README.md]: duplicated `vi`.

**Comment by Mmduh-483** [comment]
@adrianchiris

File: README.md [hunk 5]
```diff
@@ -239,6 +239,30 @@ The device plugin will initially discover all PCI network resources in the host
 4. "pfNames" - The Physical funtion name
 5. "linkTypes" - The link type of the net device associated with the PCI device.
 
+### Using UDEV Rules for changing NIC's name:
+
+UDEV rules allow the user to change the NIC's name every time the node get rebooted:
```
Comment (adrianchiris) [inline README.md]: lets add a preface to this section:

To allow "pfNames" selector to consistently select a specific device across nodes it is required to define a fixed network device name.

UDEV rules allow the user to change the NIC's network device name ....

File: README.md [hunk 6]
```diff
@@ -239,6 +239,30 @@ The device plugin will initially discover all PCI network resources in the host
 4. "pfNames" - The Physical funtion name
```
Comment (adrianchiris) [inline README.md]: nit: function net device name

not part of this commit :) but maybe a followup for this  ?

File: README.md [hunk 7]
```diff
@@ -239,6 +239,30 @@ The device plugin will initially discover all PCI network resources in the host
...
 5. "linkTypes" - The link type of the net device associated with the PCI device.
 
+### Using UDEV Rules for changing NIC's name:
```
Comment (adrianchiris) [inline README.md]: NIC's network device name

File: README.md [hunk 8]
```diff
@@ -239,6 +239,30 @@ The device plugin will initially discover all PCI network resources in the host
...
+
+UDEV rules allow the user to change the NIC's name every time the node get rebooted:
+
+1. Add a new rule:
+```
+# vi /etc/udev/rules.d/new.rules
```
Comment (adrianchiris) [inline README.md]: rules file naming convention is to name the file starting with a priority e.g : 90-netnames.rules

File: README.md (hunk 6 above)
Comment (Mmduh-483) [reply README.md]: I will consider changing it in a followup commit

**Comment by ahalim-intel** [comment]
Thanks for this PR. This is very useful, specially for "pfNames" selector. However, can we move this "how to" inside the docs/ directory in its own file and then redirect users with link to it?

**Pull request merged**